crates/driver/src/domain/competition/solution/mod.rs

```diff
@@ -246,6 +250,60 @@ impl Solution {
             Err(VerificationError::Internalization)
         }
     }
+
+    /// The clearing prices, represented as a list of assets. If there are any
+    /// orders which buy ETH, this will contain the correct ETH price.
+    pub fn prices(&self) -> Result<Vec<eth::Asset>, Error> {
+        let prices = self
+            .prices
+            .0
+            .iter()
+            .map(|(&token, &amount)| eth::Asset { token, amount });
+
+        // The solution contains an order which buys ETH. Solvers only produce solutions
+        // for ERC20 tokens, while the driver adds special [`Interaction`]s to
+        // wrap/unwrap the ETH tokens into WETH, and sends orders to the solver with
+        // WETH instead of ETH. Once the driver receives the solution which fulfills an
+        // ETH order, a clearing price for ETH needs to be added, equal to the
+        // WETH clearing price.
+        if self.user_trades().any(|trade| trade.order.buys_eth()) {
+            // If no order buys WETH, the WETH price is not necessary, only the ETH price is
+            // needed. Remove the unneeded WETH price, which slightly reduces gas used by
+            // the settlement.
+            let mut prices = if self
+                .user_trades()
+                .all(|trade| trade.order.buy.token != self.weth.0)
```

Comment:
No order can buy or sell WETH. You can have a settlement with:
- Order 1: sell WETH buy DAI
- Order 2: sell DAI  buy ETH

Which would require all three prices but get filtered out here.

Reply:
Fixed in 611791e6fb0c1800878ed1735830699c6b1e77a9